### DISCUSSION TITLE
Not easy to use Filament npm package (with examples on why)

### DISCUSSION THREAD

**Discussion by azhong-git** [google/filament#6596] [Q&A] [answered]
I understand that right now, the preferred way for loading Filament for web is to include as `<script>`. That said, for those who use npm package manager (e.g. for React or Angular), the [Filament npm package](https://www.npmjs.com/package/filament) is not easy to use and requires a few hack steps.

1. Custom webpack to `copy-webpack-plugin` `filament.wasm` in the same directory as `filament.js`
2. Custom webpack to include external Node libraries like `fs`, `crypto` and `path` despite not used by browser
3. Setting `Filament` from function to object does not work for the `import` workflow (`Filament` will still be a function and not an usable object with loaded Filament modules like `.Engine`) https://github.com/google/filament/blob/9aced8e9b8f568cab1cbd943ff35ee104a6a7bc7/web/filament-js/wasmloader.js#L71-L74

Any advice on above? Is there any plan to make the npm package usable?

Related:
https://github.com/google/filament/issues/719
https://github.com/google/filament/pull/590
https://github.com/google/filament/issues/5097
https://github.com/google/filament/pull/2913

**Comment by bejado** [comment, marked as answer]
I agree that the current state of the npm package makes things difficult. The ability to use an `import` statement is something I want to look into, but haven't had the bandwidth yet. As those comments allude to, the Emscripten machinery makes this more complicated.

**Reply by azhong-git** [reply]
Thank you for providing context. Looking forward in future if more improvements can come to the web side of things especially for React or Angular developers. Another interesting area for filament web IMO is Flutter with native support for C / web assembly.
